### PR TITLE
Remove deprecated Dashboard V3 files

### LINKED ISSUE #2522: Remove DashboardV3 after September 1st, 2020
After September 1st, 2020 the DashboardV3 will be deprecated and not shown in the app anymore.
We should all the code associated with it.

Ref: https://github.com/woocommerce/woocommerce-ios/pull/2521

### PR DESCRIPTION
**Pull request by Ecarrion**
closes https://github.com/woocommerce/woocommerce-ios/issues/2522

# Why
Since the 1st of September, we no longer support StatsV3 for the dashboard, instead, a deprecated stats screen will be shown asking the user to upgrade to the newest version of Woo.

This PR removes all(hopefully) the code related to StatsV3 since it won't be used anymore.

# How
- Removed unused code, classes, storyboards, extensions, tests, and core data entities.
- Removed sample `jsons` that were used on tests.
- Moved a function and a struct from `StatsStore` to `StatsStoreV4`

# Testing Steps
1. Checkout `develop` branch and modify the following lines of code to be able to use the old stats in the app.
On `DashboardUIFactory.swift` on line `119`
```swift
dateComponents.month = 10
```

On `StatsVersionCoordinator.swift` on line `45`
```swift
let nextVersion: StatsVersion = .v3
```
2. Run the app and see that the old stats are presented and navigate through the dates to populate the CD store
step1
.

3. Reset local changes and checkout `issue/2522-remove-Dashboard-V3` branch and modify the following lines to make V3 the default version.
On `StatsVersionCoordinator.swift` on line `45`
```swift
let nextVersion: StatsVersion = .v3
```

4. Run the app and see that the deprecated screen is shown
step2

4. Reset changes and run the app again to see the stats v4 dashboard
step3

5. Delete the app or it will crash when you resume your normal dev work on your branch because of using an outdated core data model version.

### CODE REVIEW

File: Storage/Storage/Model/MIGRATIONS.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 This file documents changes in the WCiOS Storage data model. Please explain any changes to the data model as well as any custom migrations.
 
+## Model 30 (Release 5.0.0.0)
+- @ecarrion 2020-09-02
+- Deletes `OrderStats` entity
+- Deletes `OrderStatsItem` entity
```
Comment (jaclync) [inline Storage/Storage/Model/MIGRATIONS.md]: super nit: the previous migration notes seem to be following just the simple present tense, maybe can update "Deletes" to "Delete"?
```suggestion
- Delete `OrderStats` entity
- Delete `OrderStatsItem` entity
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Dashboard.storyboard [hunk 2]
```diff
@@ -294,88 +222,6 @@
...
-                        <outlet property="containerView" destination="r3q-NT-5cw" id="Sel-D2-Nnl"/>
-                        <outlet property="headingLabel" destination="2OX-cj-Py2" id="DfC-5v-KzI"/>
-                        <outlet property="middleBorder" destination="Een-oy-QEG" id="xLy-wT-wtq"/>
-                        <outlet property="topBorder" destination="dgI-UY-ci0" id="NmM-bH-nGb"/>
-                    </connections>
-                </viewController>
-                <placeholder placeholderIdentifier="IBFirstResponder" id="gCD-ad-GJB" userLabel="First Responder" sceneMemberID="firstResponder"/>
-            </objects>
-            <point key="canvasLocation" x="-1506" y="2532"/>
-        </scene>
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Dashboard.storyboard:378]: Nice!!

**Review by jaclync — APPROVED** [review]
LGTM! So nice seeing lots of code going away, especially from the Storyboard :shipit:

Code: ✅ 
Testing: ✅  

> Testing Steps
> 1. Checkout develop branch and modify the following lines of code to be able to use the old stats in the app.
> On DashboardUIFactory.swift on line 119
> `dateComponents.month = 9`

since Sep 1st has passed, I updated to `dateComponents.month = 10` and it worked :)

`develop` v3 | PR branch v3 | PR branch v4
-- | -- | --
Simulator Screen Shot - iPhone 11 - 2020-09-03 at 14 51 55 | Simulator Screen Shot - iPhone 11 - 2020-09-03 at 14 54 08 | Simulator Screen Shot - iPhone 11 - 2020-09-03 at 14 55 37

**Comment by Ecarrion** [comment]
Thank you a lot @jaclync for the review 🙇

**Pull request merged**